src/miner.cpp

```diff
@@ -526,6 +527,10 @@ void static BitcoinMiner(const CChainParams& chainparams)
                     // Changing pblock->nTime can change work required on testnet:
                     hashTarget.SetCompact(pblock->nBits);
                 }
+
+                /* If we updated the block above, we also have to update
+                   the auxpow object to match the new block hash.  */
+                CAuxPow::initAuxPow(*pblock);
```

Comment:
I can see the point about `nBits`, about `nTime` I in turn wonder if it would not be more efficient to change that on the aux header inside this loop and break for recreation after some time threshold? That's how it's implicitly done through RPC as well, here: https://github.com/namecoin/namecoin-core/blob/master/src/rpc/mining.cpp#L883-L884?

By all means consider this a nit.